src/DDTrace/Tracer.php

```diff
@@ -406,7 +395,9 @@ public function getTracesAsArray()
             }
 
             $tracesToBeSent[] = $traceToBeSent;
-            unset($this->traces[$traceToBeSent[0]['trace_id']]);
+            if (!(empty($traceToBeSent) || empty($this->traces))) {
```

Comment:
Since `unset()` is a language construct, it will not raise a warning for unset vars so we don't need to check for `empty($this->traces)`. But for the offset, we can check that with `isset()`. :)

```suggestion
            if (isset($traceToBeSent[0]['trace_id'])) {
```